internal/utils/pubsub/src/main/java/org/eclipse/ditto/internal/utils/pubsub/actors/Subscriber.java

```diff
@@ -99,6 +76,26 @@ public static <T> Props props(final Class<T> messageClass, final PubSubTopicExtr
         return Props.create(Subscriber.class, messageClass, topicExtractor, ackExtractor, distributedAcks);
     }
 
+    /**
+     * Choose a subscriber from the subscriber pool based on the hash code of the entity ID of the published signal.
+     *
+     * @param parentSubscriber The parent subscriber from the distributed data.
+     * @param signal The signal to be published.
+     * @param poolSize The size of the subscriber pool.
+     * @return An actor selection containing the subscriber in the pool responsible for the signal.
+     */
+    static ActorSelection chooseSubscriber(final ActorRef parentSubscriber, final PublishSignal signal,
+            final int poolSize) {
+
+        if (poolSize > 1) {
+            final int index = PubSubFactory.hashForPubSub(signal.getSignal().getEntityId()) % poolSize;
+            if (index > 0) {
+                return ActorSelection.apply(parentSubscriber, String.valueOf(index));
+            }
```

Comment:
@yufei-cai are you sure this is correct?
If you have `poolSize=4` - then the modulo `%` operation will also return `0`.
However, the "sub" actors are started starting with an index number of `1` - `4`.

So how I see it, when the modulo is `0`, the `parentSusbcriber` will be selected - and the "sub" subscriber with index number `4` will never be selected.

We could solve whit by adding `+1` to the `final int index` after calculating the modulo.